src/Adapter/MySQL.php

```diff
@@ -321,22 +326,98 @@ private function computeOrderByField(array $filterToTableMapping)
             $orderField = $this->getOrderDirection() === 'asc' ? 'price_min' : 'price_max';
         }
 
-        if (array_key_exists($orderField, $filterToTableMapping)
+        $orderField = $this->computeFieldName($orderField, $filterToTableMapping);
+
+        // put some products at the end of the list
+        $orderField = $this->computeShowLast($orderField, $filterToTableMapping);
+
+        return $orderField;
+    }
+
+    /**
+     * Sort product list: InStock, OOPS with qty 0, OutOfStock
+     *
+     * @param string $orderField
+     * @param array $filterToTableMapping
+     *
+     * @return string
+     */
+    private function computeShowLast($orderField, $filterToTableMapping)
+    {
+        // allow only if feature is enabled & it is main product list query
+        if ($this->getInitialPopulation() === null
+            || empty($orderField)
+            || !Configuration::get('PS_LAYERED_FILTER_SHOW_OUT_OF_STOCK_LAST')
+        ) {
+            return $orderField;
+        }
+
+        $this->addSelectField('out_of_stock');
+
+        // order by out-of-stock last
+        $computedQuantityField = $this->computeFieldName('quantity', $filterToTableMapping);
+        $byOutOfStockLast = 'IFNULL(' . $computedQuantityField . ',0) <= 0';
```

Comment:
```suggestion
        $byOutOfStockLast = 'IFNULL(' . $computedQuantityField . ', 0) <= 0';
```
Just for having a beautiful SQL query :sweat_smile: